### DISCUSSION TITLE
Creating an "ner" object with provided "moves" not possible

### DISCUSSION THREAD

**Discussion by nikita-galileo** [explosion/spaCy#10443] [Help: Coding & Implementations]
## How to reproduce the behaviour
```
nlp = spacy.blank('en')
ner = nlp.add_pipe("ner")

move_names = ner.move_names # this will be a List[str]
moves_transition_system = ner.moves # this will be a BiluoPushDown

nlp.remove_pipe("ner")

nlp.add_pipe("ner", config={"moves": move_names}) 
# complains with: {ConfigValidationError} Config validation error ner -> moves   instance of TransitionSystem expected

nlp.add_pipe("ner", config={"moves": moves_transition_system})
# complains with: {ValueError} Found non-serializable Python object in config. Configs should only include values that can be serialized to JSON. If you need to pass models or other objects to your component, use a reference to a registered function or initialize the object in your component.
```

I'm trying to initialize a copy EntityRecognizer of an old one, but in using the component factory structure we have to provide initialization information through a `config`. I think the critical thing to provide would be the moves and the model, but not sure how to properly provide either.

**Comment by polm** [comment]
It looks like the type annotation might be off for that field - it seems like it's supposed to be a list of strings but is annotated as a TransitionSystem, which as the error points out isn't serializable (and thus shouldn't be in a config). 

To step back a little, is there a reason you aren't just [sourcing the component](https://spacy.io/usage/processing-pipelines/#sourced-components)? You should not need to copy components bit by bit like that. 

If there is some reason you are copying the component, note that if you leave moves empty it will automatically be generated based on your data, which is almost always the right thing to do. Configs in general have default values, so you only need to specify things if you don't like the defaults.

**Reply by nikita-galileo** [reply]
We're building a product that provides some spacy integration and as part of that I have a `watch/unwatch` setup around a spacy Language. So if a user does `watch(nlp)` we update components in nlp with extended functionality. When a user does `unwatch(nlp)` I would like to return the Language back to its original state. If the ner component has been trained in the meantime, I would love for the component to not lose the model weights and move_names for example. 

```
def unwatch(nlp: Language) -> None:
    """Returns spacy nlp Language component to its original unpatched state"""
    old_moves = nlp.get_pipe("ner").move_names
    old_model = nlp.get_pipe("ner").model
    nlp.remove_pipe("ner") # this 'ner' component is a patched one we want to replace with the default functionality one
    nlp.add_pipe("ner", config={"moves": old_moves, "model": old_model})
    
```

Thank you for the help!

**Reply by adrianeboyd** [reply]
I think the easiest way to do this is still with `source`:

```
original_nlp.remove_pipe("ner")
original_nlp.add_pipe("ner", source=modified_nlp)
```

The `ner` component instance should be loaded from scratch as specified in the `modified_nlp` config.
